src/player.cpp

```diff
@@ -333,32 +333,35 @@ void Player::getShieldAndWeapon(const Item*& shield, const Item*& weapon) const
 
 int32_t Player::getDefense() const
 {
-	int32_t baseDefense = 5;
-	int32_t defenseValue = 0;
-	int32_t defenseSkill = 0;
-	int32_t extraDefense = 0;
-	float defenseFactor = getDefenseFactor();
+	int32_t defenseSkill = getSkillLevel(SKILL_FIST);
+	int32_t defenseValue = 7;
 	const Item* weapon;
 	const Item* shield;
 	getShieldAndWeapon(shield, weapon);
 
 	if (weapon) {
-		defenseValue = baseDefense + weapon->getDefense();
-		extraDefense = weapon->getExtraDefense();
+		defenseValue = weapon->getDefense() + weapon->getExtraDefense();
 		defenseSkill = getWeaponSkill(weapon);
 	}
 
-	if (shield && shield->getDefense() >= defenseValue) {
-		defenseValue = baseDefense + shield->getDefense() + extraDefense;
+	if (shield) {
+		defenseValue = weapon != nullptr ? shield->getDefense() + weapon->getExtraDefense() : shield->getDefense();
 		defenseSkill = getSkillLevel(SKILL_SHIELD);
 	}
 
 	if (defenseSkill == 0) {
-		return 0;
+		switch (fightMode) {
+			case FIGHTMODE_ATTACK:
+			case FIGHTMODE_BALANCED:
+				return 1;
+
+			case FIGHTMODE_DEFENSE:
+				return 2;
+		}
 	}
 
-	defenseValue = static_cast<int32_t>(defenseValue * vocation->defenseMultiplier);
-	return static_cast<int32_t>(std::ceil((static_cast<float>(defenseSkill * (defenseValue * 0.015)) + (defenseValue * 0.1)) * defenseFactor));
+	defenseValue = static_cast<int32_t>(((((defenseSkill / 4.) + 2.23) * defenseValue) * 0.15) * getDefenseFactor());
+	return static_cast<int32_t>(static_cast<float>(defenseValue * vocation->defenseMultiplier));
```

Comment:
There are some moments when upcasting (smaller to larger int) is needed to ensure it doesn't overflow. When you divide an integer (or multiply by something <1), it will not overflow. It also probably doesn't overflow by adding small numbers such as 2.23 here, and certainly doesn't overflow after dividing by 4.

Another need for cast is when you want floating divising between two ints, where one of them must be cast to float, but you are dividing by a float here.

You will also downcast where a number comes in a larger format but the result must be shorter or to fast floor a floating point number.

This piece of code suffers none of this: ints are being automatically casted to float where needed, and there is no multiplication/sum that will ever overflow.

Also, as the return type is `int32_t`, the return is automatically casted, so `return static_cast<...>(...);` is redundant.